backend/uclapi/roombookings/helpers.py

```diff
@@ -112,23 +112,48 @@ def _parse_datetime(start_time, end_time, search_date):
 def _serialize_rooms(room_set):
     rooms = []
     for room in room_set:
-        rooms.append({
-            "roomname": room.roomname,
-            "roomid": room.roomid,
-            "siteid": room.siteid,
-            "sitename": room.sitename,
-            "capacity": room.capacity,
-            "classification": room.roomclass,
-            "automated": room.automated,
-            "location": {
-                "address": [
-                    room.address1,
-                    room.address2,
-                    room.address3,
-                    room.address4
-                ]
-            }
-        })
+        try:
+            location = Location.objects.get(siteid=room.siteid)
+        except ObjectDoesNotExist:
+            rooms.append({
+                "roomname": room.roomname,
+                "roomid": room.roomid,
+                "siteid": room.siteid,
+                "sitename": room.sitename,
+                "capacity": room.capacity,
+                "classification": room.roomclass,
+                "automated": room.automated,
+                "location": {
+                    "address": [
+                        room.address1,
+                        room.address2,
+                        room.address3,
+                        room.address4
+                    ]
+                }
+            })
+        else:
+            rooms.append({
```

Comment:
this repetition can be avoided by putting the entire insertion of coordiantes in a try except block. it will look neater and its concise.